src/Lumper.UI/ViewModels/VtfBrowser/VtfBrowserViewModel.cs

```diff
@@ -0,0 +1,125 @@
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Lumper.Lib.BSP.Struct;
+using Lumper.UI.Models;
+using Lumper.UI.Models.Matchers;
+using ReactiveUI;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.PixelFormats;
+
+namespace Lumper.UI.ViewModels.VtfBrowser;
+
+public partial class VtfBrowserViewModel : ViewModelBase
+{
+    public class Vtf : ReactiveObject
+    {
+        private Image<Rgba32>? _image;
+        private string _name = "";
+
+        public Image<Rgba32>? Image
+        {
+            get => _image;
+            set => this.RaiseAndSetIfChanged(ref _image, value);
+        }
+
+        public string Name
+        {
+            get => _name;
+            set => this.RaiseAndSetIfChanged(ref _name, value);
+        }
+    }
+
+    private double _dimensions = 128;
+
+    public double Dimensions
+    {
+        get => _dimensions;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _dimensions, value);
+            this.RaisePropertyChanged(nameof(MaxNameWidth));
+        }
+    }
+
+    public uint MaxNameWidth => (uint)_dimensions - 8;
+
+    private bool _showCubemaps;
+
+    public bool ShowCubemaps
+    {
+        get => _showCubemaps;
+        set
+        {
+            this.RaiseAndSetIfChanged(ref _showCubemaps, value);
+            this.RaisePropertyChanged(nameof(TextureBrowserItems));
+            this.RaisePropertyChanged(nameof(TexturesCount));
+        }
+    }
+
+    [GeneratedRegex(@"^((c-?\d+_-?\d+_-?\d+)|cubemapdefault)[\.hdr]{0,}\.vtf$")]
+    private static partial Regex _rgxCubemap();
+
+    private static ObservableCollection<Vtf> _textureBrowserItems =
```

Comment:
don't like that this is static .. can it not live here?

Reply:
that moving stuff to the library thing again